src/main/kotlin/engine/kafka_one/KafkaConsumerActor.kt

```diff
@@ -42,19 +42,27 @@ private fun <T, U> commitFinishedJobs(c: KafkaConsumer<T, U>,
         statuses.removeCommitted(it)
     }
 
-private fun <T, U> fetchMessagesFromKafka(c: KafkaConsumer<T, U>,
-                                          outQueue: BlockingQueue<ConsumerRecord<T, U>>,
-                                          jobStatuses: JobStatuses<T, U>) =
+private fun <T, U> fetchMessagesFromKafka(
+    c: KafkaConsumer<T, U>,
+    outQueue: BlockingQueue<ConsumerRecord<T, U>>,
+    jobStatuses: JobStatuses<T, U>
+): Pair<List<ConsumerRecord<T, U>>, JobStatuses<T, U>> =
     c.poll(POLLING_INTERVAL).let {
         if (it.count() > 0) {
             logger.info { "Adding ${it.count()} new tasks from Kafka" }
+        } else {
+            logger.debug { "Adding no new tasks from Kafka" }
         }
+        logger.debug { "Size outQueue: ${outQueue.size}, remaining capacity: ${outQueue.remainingCapacity()}" }
         outQueue.offerAll(it) to jobStatuses.addJobs(it)
     }
 
-private fun <T> BlockingQueue<T>.offerAll(xs: Iterable<T>) = xs.dropWhile { offer(it) }
+private fun <T> BlockingQueue<T>.offerAll(xs: Iterable<T>): List<T> = xs.dropWhile { offer(it) }
```

Comment:
Nice, love when people set things like this explicit.

Reply:
Agreed. Even if this is a standard Kotlin operation on a collection, explicit typing are nice.